bentoml/_internal/yatai_client/schemas.py

```diff
@@ -0,0 +1,240 @@
+from dataclasses import dataclass, field
+from datetime import datetime
+from enum import Enum
+from typing import Optional, List, Dict, Any
+
+from dataclasses_json import DataClassJsonMixin
+from marshmallow import fields
+from dateutil.parser import parse
+
+
+time_format = '%Y-%m-%d %H:%M:%S.%f'
+
+
+def datetime_encoder(time_obj: Optional[datetime]) -> Optional[str]:
+    if not time_obj:
+        return None
+    return time_obj.strftime(time_format)
+
+
+def datetime_decoder(datetime_str: Optional[str]) -> Optional[datetime]:
+    if not datetime_str:
+        return None
+    return parse(datetime_str)
+
+
+datetime_field_kwargs = dict(metadata={'dataclasses_json': {
+    'encoder': datetime_encoder,
+    'decoder': datetime_decoder,
+    'mm_field': fields.DateTime(format='iso')
+}})
+
+
+@dataclass
+class BaseSchema(DataClassJsonMixin):
+    uid: str = ''
+    created_at: datetime = field(default_factory=datetime.now, **datetime_field_kwargs)
+    updated_at: Optional[datetime] = field(default=None, **datetime_field_kwargs)
+    deleted_at: Optional[datetime] = field(default=None, **datetime_field_kwargs)
+
+
+class BaseListSchema(DataClassJsonMixin):
+    start: int
+    count: int
+    total: int
+
+
+class ResourceType(Enum):
+    USER = 'user'
+    ORG = 'organization'
+    CLUSTER = 'cluster'
+    BENTO = 'bento'
+    BENTO_VERSION = 'bento_version'
+    MODEL = 'model'
+    MODEL_VERSION = 'model_version'
+
+
+@dataclass
+class ResourceSchema(BaseSchema, DataClassJsonMixin):
+    name: str = ''
+    resource_type: ResourceType = ResourceType.USER
+
+
+@dataclass
+class LabelItemSchema(DataClassJsonMixin):
+    key: str = ''
+    value: str = ''
+
+
+@dataclass
+class UserSchema(ResourceSchema):
+    name: str = ''
+    email: str = ''
+    first_name: str = ''
+    last_name: str = ''
+
+    def get_name(self) -> str:
+        if not self.first_name and not self.last_name:
+            return self.name
+        return f'{self.first_name} {self.last_name}'.strip()
+
+
+@dataclass
+class OrganizationSchema(ResourceSchema, DataClassJsonMixin):
+    description: str = ''
+
+
+@dataclass
+class OrganizationListSchema(BaseListSchema):
+    items: List[OrganizationSchema]
+
+
+@dataclass
+class ClusterSchema(ResourceSchema):
+    description: str = None
+
+
+@dataclass
+class BentoSchema(ResourceSchema):
```

Comment:
We have plan to "flatten" the Bento & versions page right?  Let's also reflect that in the API and backend storage. There's no need for both BentoSchema and a BentoVersionSchema, it should be just one BentoSchema

Reply:
@parano I think leveling is just something to do on the front-end presentation, and blending Bento and BentoVersion together in the back-end abstraction is a very irrational approach